katsdpcontroller/generator.py

```diff
@@ -1899,6 +1902,31 @@ def _make_spectral_imager(g, config, capture_block_id, name, telstate, target_ma
                     ', '.join(target.name for target, _ in targets))
     view = telstate.view(telstate.join(capture_block_id, name))
     view['targets'] = {target.description: target_mapper(target) for target, _ in targets}
+    # While it is currently just a contiguous range, output the channels as a
+    # list to allow for applying a static mask in future.
+    view['output_channels'] = list(range(output_channels[0], output_channels[1]))
+    return data_url, nodes
+
+
+def _make_spectral_imager_report(g, config, capture_block_id, name, data_url, spectral_nodes):
+    report = SDPLogicalTask(f'spectral_image_report.{name}')
+    report.cpus = 1.0
+    # Memory is a guess - but since we don't open the chunk store it should be lightweight
+    report.mem = 4 * 1024
```

Comment:
Happy enough with a guess, but given the severity of OOM, is there any scaling in the report related to the field complexity or observation length ? (I doubt it, but just thought I would check)

Reply:
I do use katdal is used to open the observation. However, I don't access the data at all, and it's a live telstate so it won't need to copy all the raw sensor data into memory. There will certainly be some scaling with time and with channels, but it should be small since it won't scale with time × channels × baselines, just time + channels.